File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/domain/ipAllowlist/IpAllowlistExceptions.java
package br.com.deltaglobalbank.identity.domain.ipAllowlist;

public abstract sealed class IpAllowlistExceptions extends RuntimeException
    permits InvalidCidrException, CidrAlreadyExistsException,
    IpAllowlistEntryNotFoundException, IpNotAllowedException {

    protected IpAllowlistExceptions(String message) {
        super(message);
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/domain/ipAllowlist/IpAllowListExceptions.kt
package br.com.deltaglobalbank.identity.domain.ipAllowlist

sealed class IpAllowlistExceptions(message: String) : RuntimeException(message)

class InvalidCidrException : IpAllowlistExceptions("invalid_cidr")
class CidrAlreadyExistsException : IpAllowlistExceptions("cidr_already_exists")
class IpAllowlistEntryNotFoundException : IpAllowlistExceptions("ip_allowlist_entry_not_found")
class IpNotAllowedException : IpAllowlistExceptions("ip_not_allowed")